[IMP] http: Enhance the set_cookie

imanie383 opened the pull request:
Description of the issue/feature this PR addresses:

Enhance session management in an Islamic website by implementing a secondary cookie (multi_session_id). Upon user login, the system must register the multi_session_id cookie within the base domain, ensuring it functions alongside the original session cookie. Similarly, upon user logout, the multi_session_id cookie should be deleted in the same manner as the primary session cookie to maintain consistency and security


---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

CarmenMiranda commented on odoo/http.py:
```diff
@@ -1210,6 +1210,11 @@ def set_cookie(self, key, value='', max_age=None, expires=None, path='/', domain
+        if request.session.get("multi_login"):
+            expires = None
+            max_age = None
+        elif key == 'session_id' and not domain and (subdomain := request.env['ir.http']._get_subdomain(domain)):
```
The subdomain here is not for multi-login? The parameters used in `_get_subdomain` are for multi

imanie383 replied:
https://gitlab.com/-/project/47072011/uploads/cd8024d1b83fdef8dd4143382e2e3a76/2025-03-24_18-36-17.mp4

What I'm doing here is saving a session-type cookie on the configured domain, as long as the current instance is a base subdomain. This way, when accessing from another subdomain, it can access this cookie and log in using the information stored in it.

CarmenMiranda replied:
Ohh, what I meant was that if it wasn't needed when "multi_login" is in the session, because I thought that in those cases the multi_login was there 🤔

imanie383 committed:
[IMP] http: Enhance the set_cookie

Enhance the set_cookie flow to ensure that only the main session has the capability to update it. Secondary website instances (referred to as "slaves") should instead use a session-type cookie, preventing them from modifying the primary session's cookie. This enhancement aims to strengthen session integrity and prevent unintended overwrites or conflicts between the main session and the linked websites.
